torch/jit/annotations.py

```diff
@@ -240,6 +242,22 @@ def as_ann(ann):
     return arg_types, return_type
 
 
+# Finds common type for enum values belonging to an Enum class. If not all
+# values have the same type, AnyType is returned.
+def get_enum_value_type(e: enum.Enum, loc):
+    enum_values = list(e)
+    if not enum_values:
+        raise ValueError("No enum values defined for: '{}'".format(e.__class__))
+
+    types = set([type(v.value) for v in enum_values])
+
+    if len(types) != 1:
+        raise NotImplementedError(
+            "Enum class with heterogenous value types is not supported")
```

Comment:
It'd be better to use https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/core/jit_type.h#L1513 so that we get a good error message here ("Type str did not match type int...") . you'd need to convert the types to jit type first with ann_to_type.

Reply:
Done.